### ISSUE TITLE
goto interacts with block inappropriately 

### ISSUE THREAD

**Issue by SylveonBottle** [cabaletta/baritone#400]
copy/paste from discord

let's not have goto bed result in bariton trying to destroy beds. it doesn't need go inside of the bed, just close enough to right-click it. Infact bariton can stop pathing once it's able to right-click the bed/chest
just got stuck in a loop where bariton refused to "accept" that it was already at the chest, so it kept rightclicking?
so basically if goto signals right-clicking / interacting with a block (chest/bed/etc), assume the user only needs to interact with it - not stand right up against it or go inside of it. User probably doesn't even need to be right next to it, so just finish pathing once we're "close enough" for interaction.
or, rather, after the interaction itself. Also try to interact when we're in range instead of once we're finished pathing
this may or may not be worth the time to watch on anything but x2 speed https://youtu.be/v-v2qJBBt-Q

TL;DR:
-If specified X/Y/Z has an intractable block, let's assume the user wants to interact with it and not break / go inside of it, and just open/interact with it whenever we get close enough.

**Comment by leijurv** [comment]
Going into a bed and breaking the bed is fixed, I just yesterday committed a fix for the repeated right clicking of containers (like chests). https://github.com/cabaletta/baritone/commit/166eb97c3fd882fd25cc14702bff0482b983887f
